### ISSUE TITLE
Adding cp.squeeze

### ISSUE THREAD

**Issue by Transurgeon** [cvxpy/cvxpy#2290]
**Is your feature request related to a problem? Please describe.**
Missing squeeze operation for full compatibility with NumPy. This currently has the same functionality as flatten since cvxpy expressions are limited to 2-dimensions. However with the addition of nd-expressions, the squeeze operation could be helpful to ensure shape consistency. 

**Describe the solution you'd like**
Implement an equivalent operation to [np.squeeze](https://numpy.org/doc/stable/reference/generated/numpy.squeeze.html) in cvxpy.

**Comment by yemine0x01** [comment]
Is this still of interest? I'd like to give it a try.

**Comment by SteveDiamond** [comment]
It's still of interest. Contributions are very helpful.

**Comment by yemine0x01** [comment]
I’ve been thinking about this and was considering implementing a `Squeeze` class as a subclass of `AffAtom`, similar to how the `reshape` class is structured. Does this approach make sense, or am I heading in the wrong direction?

**Comment by Transurgeon** [comment]
@eminyouskn yes I think your approach makes sense! In fact, I think squeeze can even just be a wrapper around reshape? But you would need to compute the output shape by iterating over all dimensions and removing the ones.

**Comment by yemine0x01** [comment]
@Transurgeon I implement it as you suggested. Thanks.

**Comment by yemine0x01** [comment]
I think we can close this issue.

**Comment by Transurgeon** [comment]
thanks @eminyouskn nice work!
